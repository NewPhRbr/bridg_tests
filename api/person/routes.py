from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import orm

from api.db import get_db
from api.person import schema
from api.util import setattrs
from umdb import person as model

router = APIRouter(prefix="/persons", tags=["persons"])


@router.get("/", response_model=list[schema.Person])
def index(db: orm.Session = Depends(get_db)):
    return db.query(model.Person).all()


@router.post("/", response_model=schema.Person)
def create(body: schema.PersonCreate, db: orm.Session = Depends(get_db)):
    name = setattrs(model.Name(), body.name.model_dump())
    person = setattrs(model.Person(), body.model_dump(exclude={"name"}))
    person.name.append(name)
    db.add(person)
    db.commit()
    db.refresh(person)
    return person


@router.get("/{id}", response_model=schema.Person)
def read(id: int, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    return person


@router.patch("/{id}", response_model=schema.Person)
def update(id: int, body: schema.PersonUpdate, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    person = setattrs(person, body.model_dump())
    db.add(person)
    db.commit()
    db.refresh(person)
    return person


@router.delete("/{id}")
def delete(id: int, db: orm.Session = Depends(get_db)):
    person = db.query(model.Person).where(model.Person.id == id).first()
    if person is None:
        raise HTTPException(404)
    db.delete(person)
    db.commit()
