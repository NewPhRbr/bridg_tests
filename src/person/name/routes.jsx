import * as personAPI from "../api";
import * as nameAPI from "./api";
import Edit from "./edit";

export default [
  {
    path: ":nameId",
    children: [
      {
        path: "edit",
        Component: Edit,
        loader: async ({ params: { personId, nameId } }) => ({
          person: await personAPI.find(personId).then((x) => x.json()),
          name: await nameAPI.find(personId, nameId).then((x) => x.json()),
        }),
      },
    ],
  },
];
