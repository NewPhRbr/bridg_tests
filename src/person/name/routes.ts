import * as api from "./api";
import Edit from "./edit";

export default [
  {
    path: ":nameId",
    children: [
      {
        path: "edit",
        Component: Edit,
        loader: ({ params: { personId, nameId } }) => api.find(personId, nameId),
        handle: {
          breadcrumb: "Rename",
        },
      },
    ],
  },
];
