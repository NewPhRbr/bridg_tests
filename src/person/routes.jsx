import * as api from "./api";
import Index from "./index";
import Edit from "./edit";

export default [
  {
    path: "",
    element: <Index />,
  },
  {
    path: ":id",
    loader: ({ params: { id } }) => api.find(id),
    Component: Edit,
  },
];
