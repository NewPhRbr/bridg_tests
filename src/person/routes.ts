import * as api from "./api";
import Edit from "./edit";
import Index from "./index";
import nameRoutes from "./name/routes";
import New from "./new";
import Show from "./show";

export default [
  {
    handle: {
      breadcrumb: "Persons",
    },
    children: [
      {
        path: "",
        Component: Index,
      },
      {
        path: "new",
        Component: New,
        handle: {
          breadcrumb: "New",
        },
      },
      {
        id: "person",
        path: ":personId",
        loader: ({ params: { personId } }: any) => api.find(personId),
        handle: {
          breadcrumb: (data) => data.name?.full,
        },
        children: [
          {
            path: "",
            Component: Show,
            loader: ({ params: { personId } }: any) => api.find(personId),
          },
          {
            path: "edit",
            Component: Edit,
            loader: ({ params: { personId } }: any) => api.find(personId),
            handle: {
              breadcrumb: "Edit",
            },
          },
          {
            path: "name",
            children: nameRoutes,
          },
        ],
      },
    ],
  },
];
