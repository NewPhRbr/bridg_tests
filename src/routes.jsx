import Layout from "./Layout";
import personRoutes from "./person/routes";

export default [
  {
    path: "/",
    element: <Layout />,
    children: [
      {
        path: "",
        element: <div>Works</div>,
      },
      {
        path: "/persons",
        children: personRoutes,
      },
    ],
  },
];
