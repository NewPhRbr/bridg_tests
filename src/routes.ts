import Layout from "./Layout";
import personRoutes from "./person/routes";

export default [
  {
    path: "/",
    Component: Layout,
    children: [
      {
        path: "",
      },
      {
        path: "/persons",
        children: personRoutes,
      },
    ],
  },
];
