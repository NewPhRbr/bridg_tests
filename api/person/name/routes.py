from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy import orm

from api.db import get_db
from api.person.name import schema
from api.util import setattrs
from umdb import common as model

router = APIRouter(prefix="/persons/{person_id}/names", tags=["names"])


@router.get("/", response_model=list[schema.Name])
def index(person_id: int, db: orm.Session = Depends(get_db)):
    return db.query(model.Name).where(model.Name.person_id == person_id).all()


@router.get("/{id}", response_model=schema.Name)
def find(person_id: int, id: int, db: orm.Session = Depends(get_db)):
    name = db.query(model.Name).where(model.Name.id == id).first()
    if name is None:
        raise HTTPException(404)
    return name


@router.patch("/{id}", response_model=schema.Name)
def update(
    person_id: int,
    id: int,
    body: schema.NameUpdate,
    db: orm.Session = Depends(get_db),
):
    name = db.query(model.Name).where(model.Name.id == id).first()
    if name is None:
        raise HTTPException(404)
    name = setattrs(name, body.model_dump())
    db.add(name)
    db.commit()
    db.refresh(name)
    return name
