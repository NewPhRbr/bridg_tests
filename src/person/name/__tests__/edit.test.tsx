import { render, screen } from "@testing-library/react";
import { HttpResponse, http } from "msw";
import { setupServer } from "msw/node";
import { RouterProvider, createMemoryRouter } from "react-router-dom";
import { afterAll, afterEach, beforeAll, expect, test } from "vitest";
import routes from "../../../routes";

const URL = "/persons/1/name/2/edit";

const PERSON = {
  sex: "M",
  birth_date: "2024-05-21",
  death_date: "2024-05-21",
  death_date_estimated_indicator: true,
  death_indicator: true,
  id: 1,
  primary_name: {
    use: "string",
    family: "John",
    given: "Doe",
    middle: "",
    patronymic: "",
    prefix: "",
    suffix: "",
    id: 2,
    full: "Doe John",
  },
};

const NAME = {
  use: "string",
  family: "John",
  given: "Doe",
  middle: "",
  patronymic: "",
  prefix: "",
  suffix: "",
  id: 2,
  full: "Doe John",
};

const server = setupServer(
  http.get("/api/persons/1", () => HttpResponse.json(PERSON)),
  http.get("/api/persons/1/names/2", () => HttpResponse.json(NAME))
);

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

test("person's name edit page renders correctly", async () => {
  const router = createMemoryRouter(routes, { initialEntries: [URL] });
  const tree = render(<RouterProvider router={router} />);
  await screen.findAllByText("Doe John");
  expect(tree).toMatchSnapshot();
});
