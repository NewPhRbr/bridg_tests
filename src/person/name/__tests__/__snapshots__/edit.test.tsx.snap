// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`person's name edit page renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="chakra-stack css-24ivmg"
      >
        <div
          class="chakra-stack css-1b5q7m6"
        >
          <b
            class="chakra-text css-1fqaqvh"
          >
            <a
              class=""
              href="/"
            >
              umdb
            </a>
          </b>
          <div
            class="chakra-stack css-1igwmid"
          >
            <a
              aria-current="page"
              class="active"
              href="/persons"
            >
              <p
                class="chakra-text css-19ner23"
              >
                Persons
              </p>
            </a>
          </div>
        </div>
        <div
          class="css-139mjay"
        >
          <nav
            aria-label="breadcrumb"
            class="chakra-breadcrumb css-0"
          >
            <ol
              class="chakra-breadcrumb__list css-70qvj9"
            >
              <li
                class="chakra-breadcrumb__list-item css-18biwo"
              >
                <a
                  class="chakra-breadcrumb__link css-0"
                  href="/persons/"
                >
                  Persons
                </a>
                <span
                  class="css-t4q1nq"
                  role="presentation"
                >
                  /
                </span>
              </li>
              <li
                class="chakra-breadcrumb__list-item css-18biwo"
              >
                <a
                  class="chakra-breadcrumb__link css-0"
                  href="/persons/1"
                >
                  Doe John
                </a>
                <span
                  class="css-t4q1nq"
                  role="presentation"
                >
                  /
                </span>
              </li>
              <li
                class="chakra-breadcrumb__list-item css-18biwo"
              >
                <a
                  aria-current="page"
                  class="chakra-breadcrumb__link css-0"
                  href="/persons/1/name/2/edit"
                >
                  Rename
                </a>
              </li>
            </ol>
          </nav>
        </div>
        <div
          class="css-139mjay"
        >
          <div
            class="chakra-stack css-24ivmg"
          >
            <div
              class="css-k008qs"
            >
              <p
                class="chakra-text css-1n9r2z"
              >
                Doe John
              </p>
              <div
                class="css-17xejub"
              />
              <button
                class="chakra-button css-4xx2wk"
                form="name-form"
                type="submit"
              >
                Save
              </button>
            </div>
            <div
              class="chakra-stack css-24ivmg"
            >
              <div
                class="chakra-card css-2dlz7"
              >
                <div
                  class="chakra-card__header css-0"
                >
                  <h2
                    class="chakra-heading css-0"
                  >
                    Edit information
                  </h2>
                </div>
                <div
                  class="chakra-card__body css-0"
                >
                  <form
                    id="name-form"
                  >
                    <div
                      class="chakra-stack css-24ivmg"
                    >
                      <div
                        class="chakra-stack css-1igwmid"
                      >
                        <div
                          class="chakra-form-control css-68zbsl"
                          role="group"
                        >
                          <label
                            class="chakra-form__label css-uvho5g"
                            for="field-:r0:"
                            id="field-:r0:-label"
                          >
                            Family
                          </label>
                          <input
                            class="chakra-input css-0"
                            id="field-:r0:"
                            name="family"
                          />
                        </div>
                        <div
                          class="chakra-form-control css-68zbsl"
                          role="group"
                        >
                          <label
                            class="chakra-form__label css-uvho5g"
                            for="field-:r1:"
                            id="field-:r1:-label"
                          >
                            Middle
                          </label>
                          <input
                            class="chakra-input css-0"
                            id="field-:r1:"
                            name="middle"
                          />
                        </div>
                        <div
                          class="chakra-form-control css-68zbsl"
                          role="group"
                        >
                          <label
                            class="chakra-form__label css-uvho5g"
                            for="field-:r2:"
                            id="field-:r2:-label"
                          >
                            Given
                          </label>
                          <input
                            class="chakra-input css-0"
                            id="field-:r2:"
                            name="given"
                          />
                        </div>
                      </div>
                      <details>
                        <summary>
                          Extra
                        </summary>
                        <div
                          class="chakra-stack css-1g2gf02"
                        >
                          <div
                            class="chakra-stack css-1igwmid"
                          >
                            <div
                              class="chakra-form-control css-rgf84b"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                for="field-:r3:"
                                id="field-:r3:-label"
                              >
                                Prefix
                              </label>
                              <input
                                class="chakra-input css-0"
                                id="field-:r3:"
                                name="prefix"
                              />
                            </div>
                            <div
                              class="chakra-form-control css-rgf84b"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                for="field-:r4:"
                                id="field-:r4:-label"
                              >
                                Suffix
                              </label>
                              <input
                                class="chakra-input css-0"
                                id="field-:r4:"
                                name="suffix"
                              />
                            </div>
                          </div>
                          <div
                            class="chakra-stack css-1igwmid"
                          >
                            <div
                              class="chakra-form-control css-68zbsl"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                for="field-:r5:"
                                id="field-:r5:-label"
                              >
                                Patronymic
                              </label>
                              <input
                                class="chakra-input css-0"
                                id="field-:r5:"
                                name="patronymic"
                              />
                            </div>
                          </div>
                          <div
                            class="chakra-stack css-1igwmid"
                          >
                            <div
                              class="chakra-form-control css-68zbsl"
                              role="group"
                            >
                              <label
                                class="chakra-form__label css-uvho5g"
                                for="field-:r6:"
                                id="field-:r6:-label"
                              >
                                Use
                              </label>
                              <div
                                class="chakra-select__wrapper css-42b2qy"
                              >
                                <select
                                  class="chakra-select css-1ik61og"
                                  id="field-:r6:"
                                  name="use"
                                >
                                  <option
                                    value=""
                                  />
                                  <option
                                    value="official"
                                  >
                                    official
                                  </option>
                                  <option
                                    value="nickname"
                                  >
                                    nickname
                                  </option>
                                </select>
                                <div
                                  class="chakra-select__icon-wrapper css-162mkon"
                                >
                                  <svg
                                    aria-hidden="true"
                                    class="chakra-select__icon"
                                    focusable="false"
                                    role="presentation"
                                    style="width: 1em; height: 1em; color: currentColor;"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                                      fill="currentColor"
                                    />
                                  </svg>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </details>
                    </div>
                    <button
                      class="chakra-button css-1bpmanl"
                      type="submit"
                    >
                      Save
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-24ivmg"
    >
      <div
        class="chakra-stack css-1b5q7m6"
      >
        <b
          class="chakra-text css-1fqaqvh"
        >
          <a
            class=""
            href="/"
          >
            umdb
          </a>
        </b>
        <div
          class="chakra-stack css-1igwmid"
        >
          <a
            aria-current="page"
            class="active"
            href="/persons"
          >
            <p
              class="chakra-text css-19ner23"
            >
              Persons
            </p>
          </a>
        </div>
      </div>
      <div
        class="css-139mjay"
      >
        <nav
          aria-label="breadcrumb"
          class="chakra-breadcrumb css-0"
        >
          <ol
            class="chakra-breadcrumb__list css-70qvj9"
          >
            <li
              class="chakra-breadcrumb__list-item css-18biwo"
            >
              <a
                class="chakra-breadcrumb__link css-0"
                href="/persons/"
              >
                Persons
              </a>
              <span
                class="css-t4q1nq"
                role="presentation"
              >
                /
              </span>
            </li>
            <li
              class="chakra-breadcrumb__list-item css-18biwo"
            >
              <a
                class="chakra-breadcrumb__link css-0"
                href="/persons/1"
              >
                Doe John
              </a>
              <span
                class="css-t4q1nq"
                role="presentation"
              >
                /
              </span>
            </li>
            <li
              class="chakra-breadcrumb__list-item css-18biwo"
            >
              <a
                aria-current="page"
                class="chakra-breadcrumb__link css-0"
                href="/persons/1/name/2/edit"
              >
                Rename
              </a>
            </li>
          </ol>
        </nav>
      </div>
      <div
        class="css-139mjay"
      >
        <div
          class="chakra-stack css-24ivmg"
        >
          <div
            class="css-k008qs"
          >
            <p
              class="chakra-text css-1n9r2z"
            >
              Doe John
            </p>
            <div
              class="css-17xejub"
            />
            <button
              class="chakra-button css-4xx2wk"
              form="name-form"
              type="submit"
            >
              Save
            </button>
          </div>
          <div
            class="chakra-stack css-24ivmg"
          >
            <div
              class="chakra-card css-2dlz7"
            >
              <div
                class="chakra-card__header css-0"
              >
                <h2
                  class="chakra-heading css-0"
                >
                  Edit information
                </h2>
              </div>
              <div
                class="chakra-card__body css-0"
              >
                <form
                  id="name-form"
                >
                  <div
                    class="chakra-stack css-24ivmg"
                  >
                    <div
                      class="chakra-stack css-1igwmid"
                    >
                      <div
                        class="chakra-form-control css-68zbsl"
                        role="group"
                      >
                        <label
                          class="chakra-form__label css-uvho5g"
                          for="field-:r0:"
                          id="field-:r0:-label"
                        >
                          Family
                        </label>
                        <input
                          class="chakra-input css-0"
                          id="field-:r0:"
                          name="family"
                        />
                      </div>
                      <div
                        class="chakra-form-control css-68zbsl"
                        role="group"
                      >
                        <label
                          class="chakra-form__label css-uvho5g"
                          for="field-:r1:"
                          id="field-:r1:-label"
                        >
                          Middle
                        </label>
                        <input
                          class="chakra-input css-0"
                          id="field-:r1:"
                          name="middle"
                        />
                      </div>
                      <div
                        class="chakra-form-control css-68zbsl"
                        role="group"
                      >
                        <label
                          class="chakra-form__label css-uvho5g"
                          for="field-:r2:"
                          id="field-:r2:-label"
                        >
                          Given
                        </label>
                        <input
                          class="chakra-input css-0"
                          id="field-:r2:"
                          name="given"
                        />
                      </div>
                    </div>
                    <details>
                      <summary>
                        Extra
                      </summary>
                      <div
                        class="chakra-stack css-1g2gf02"
                      >
                        <div
                          class="chakra-stack css-1igwmid"
                        >
                          <div
                            class="chakra-form-control css-rgf84b"
                            role="group"
                          >
                            <label
                              class="chakra-form__label css-uvho5g"
                              for="field-:r3:"
                              id="field-:r3:-label"
                            >
                              Prefix
                            </label>
                            <input
                              class="chakra-input css-0"
                              id="field-:r3:"
                              name="prefix"
                            />
                          </div>
                          <div
                            class="chakra-form-control css-rgf84b"
                            role="group"
                          >
                            <label
                              class="chakra-form__label css-uvho5g"
                              for="field-:r4:"
                              id="field-:r4:-label"
                            >
                              Suffix
                            </label>
                            <input
                              class="chakra-input css-0"
                              id="field-:r4:"
                              name="suffix"
                            />
                          </div>
                        </div>
                        <div
                          class="chakra-stack css-1igwmid"
                        >
                          <div
                            class="chakra-form-control css-68zbsl"
                            role="group"
                          >
                            <label
                              class="chakra-form__label css-uvho5g"
                              for="field-:r5:"
                              id="field-:r5:-label"
                            >
                              Patronymic
                            </label>
                            <input
                              class="chakra-input css-0"
                              id="field-:r5:"
                              name="patronymic"
                            />
                          </div>
                        </div>
                        <div
                          class="chakra-stack css-1igwmid"
                        >
                          <div
                            class="chakra-form-control css-68zbsl"
                            role="group"
                          >
                            <label
                              class="chakra-form__label css-uvho5g"
                              for="field-:r6:"
                              id="field-:r6:-label"
                            >
                              Use
                            </label>
                            <div
                              class="chakra-select__wrapper css-42b2qy"
                            >
                              <select
                                class="chakra-select css-1ik61og"
                                id="field-:r6:"
                                name="use"
                              >
                                <option
                                  value=""
                                />
                                <option
                                  value="official"
                                >
                                  official
                                </option>
                                <option
                                  value="nickname"
                                >
                                  nickname
                                </option>
                              </select>
                              <div
                                class="chakra-select__icon-wrapper css-162mkon"
                              >
                                <svg
                                  aria-hidden="true"
                                  class="chakra-select__icon"
                                  focusable="false"
                                  role="presentation"
                                  style="width: 1em; height: 1em; color: currentColor;"
                                  viewBox="0 0 24 24"
                                >
                                  <path
                                    d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                                    fill="currentColor"
                                  />
                                </svg>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </details>
                  </div>
                  <button
                    class="chakra-button css-1bpmanl"
                    type="submit"
                  >
                    Save
                  </button>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
